android/app/src/main/java/app/organicmaps/sdk/sync/SyncState.java

```diff
@@ -0,0 +1,121 @@
+package app.organicmaps.sdk.sync;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+
+public class SyncState
+{
+  private static final String PREF_CHECKSUM_CACHE_PREFIX = "sync-state-";
+
+  private static final String PREF_METADATA_PREFIX = "sync-metadata-";
+  private static final String PREF_KEY_CHANGED_FILES = "changed";
+  private static final String PREF_KEY_FILES_CACHE_OUTDATED = "changedFilesOutdated";
+  private static final String PREF_KEY_ROOT_DIR_STATE = "root";
+
+  private final SharedPreferences mMetadataPrefs;
+  private final SharedPreferences mChecksumPrefs;
```

Comment:
There is `SyncPrefs` class that I read as `SyncPreferences` and according to its name it works with shared preferences. Why do you use additional `SharedPreferences` here?

Reply:
And what is the point in having so many shared preferences? Can't it be just one shared prefs?